middleware/kubernetes/kubernetes.go

```diff
@@ -227,40 +269,42 @@ func (k *Kubernetes) Records(name string, exact bool) ([]msg.Service, error) {
 	return records, nil
 }
 
-func (k *Kubernetes) getRecordsForServiceItems(serviceItems []*api.Service, zone string) []msg.Service {
+func endpointHostname(addr api.EndpointAddress) string {
+	if addr.Hostname != "" {
+		return strings.ToLower(addr.Hostname)
+	} else {
+		if strings.Contains(addr.IP, ".") {
+			return strings.Replace(addr.IP, ".", "-", -1)
+		}
+		if strings.Contains(addr.IP, ":") {
```

Comment:
this will convert `fe80::00` to `fe80--00`, or is addr.IP always a complete v6 address? Also https://github.com/miekg/coredns/pull/452 has similar code, might be worth abstracting in the future.